src/pollution_model/run_medusa_2.py

```diff
@@ -1,474 +1,520 @@
+    for i in range(len(all_roads)):
+        surface_area = float(all_roads.iloc[i]["SurfaceArea"])
+        surface_type = all_roads.iloc[i]["SurfaceType"]
+        curr_tss = compute_tss_roof_road(surface_area=surface_area, antecedent_dry_days=antecedent_dry_days,
+                                         average_rain_intensity=average_rain_intensity, event_duration=event_duration,
+                                         surface_type=surface_type)
+
+        curr_total_copper, curr_total_zinc = total_metal_load_road_carpark(curr_tss)
+        curr_dissolved_copper, curr_dissolved_zinc = dissolved_metal_load(total_copper_load=curr_total_copper,
+                                                                          total_zinc_load=curr_total_zinc,
+                                                                          surface_type=surface_type)
+
+        all_roads.iloc[i] = [all_roads.iloc[i]["Index"], surface_area, surface_type, curr_tss,
+                             curr_total_copper, curr_total_zinc, curr_dissolved_copper, curr_dissolved_zinc]
+
+    # all_result = xr.merge(all_roads, all_buildings)
+    return gpd.GeoDataFrame(pd.concat([all_roads, all_buildings], ignore_index=True))
+
+
+def store_pollution_model_in_database(engine: Engine, results: gpd.GeoDataFrame, scenario_id: int):
+    """
+    Appends the details of which buildings are flooded for a given flood_model_id to the database
+
+    Parameters
+    ----------
+    engine: Engine
+        The sqlalchemy database connection engine
+    results : gpd.GeoDataFrame
+        GeoDataFrame containing a mapping of geospatial features (roads, roofs) to their MEDUSA2.0 outputs for the
+        current model run
+    scenario_id : int
+        The id of the current medusa2.0 model run, to associate with the results.
+
+    Returns
+    -------
+    None
+        This function does not return anything
+    """
+    results["scenario_id"] = scenario_id
+    results.set_index("Index", inplace=True)
+    results.to_sql("medusa2_model_output", engine, if_exists="append", index=True)
+    # Create geoserver endpoints for database views if they do not already exist I noticed this was in the code for a
+    # similar function, so I've added it. Please comment if this is necessary or not @Luke
+    create_building_database_views_if_not_exists()
+
+
+def get_next_scenario_id(engine):
+    with engine.begin() as conn:
+        result = conn.execute("SELECT MAX(scenario_id) FROM medusa2_model_output").fetchone()[0]
+        return (result or 0) + 1
+
+
+def main(selected_polygon_gdf: gpd.GeoDataFrame,
+         log_level: LogLevel = LogLevel.DEBUG,
+         antecedent_dry_days: float = 1,
+         average_rain_intensity: float = 10000,
+         event_duration: float = 5,
+         rainfall_ph: float = 7):
+    """
+    Generate pollution model output for the requested catchment area, and incorporate the model output to GeoServer
+    for visualization.
```

Comment:
```suggestion
    Generate pollution model output for the requested catchment area, and incorporate the model output to GeoServer
    for visualization.
```
You don't need to worry about geoserver for this issue, it can be covered in #160.
```suggestion
    Generate pollution model output for the requested catchment area, and save result to database.
```